src/librbd/Journal.cc

```diff
@@ -802,6 +802,44 @@ uint64_t Journal<I>::append_write_event(uint64_t offset, size_t length,
 }
 
 template <typename I>
+uint64_t Journal<I>::append_compare_and_write_event(uint64_t offset, size_t length,
+                                        const bufferlist &cmp_bl, const bufferlist &bl,
+                                        const IOObjectRequests &requests,
+                                        bool flush_entry) {
+  assert(m_max_append_size > journal::AioCompareAndWriteEvent::get_fixed_size());
```

Comment:
This max size stuff can be removed -- it's an unnecessary complication for an op that will always be essentially just a sector in length. The "max" size is just a soft limit to ensure we can control the memory usage of rbd-mirror daemon if clients are performing huge write operations (e.g. lots of 128MB writes).